pkg/clustermanager/providerclient/client_interface.go

```diff
@@ -91,13 +93,23 @@ type WatchEvent struct {
 // The provider is responsible for discovering and managing the lifecycle of each
 // cluster.
 type ProviderClient interface {
-	Create(ctx context.Context, name string, opts Options) (LogicalClusterInfo, error)
+	Create(ctx context.Context, name string, opts Options) error
 	Delete(ctx context.Context, name string, opts Options) error
 
 	// List returns a list of logical clusters.
 	// This method is used to discover the initial set of logical clusters
 	// and to refresh the list of logical clusters periodically.
-	List() ([]string, error)
+	List(ctx context.Context) ([]LogicalClusterInfo, error)
+
+	// List returns a list of logical cluster names.
+	// This method is used to discover the initial set of logical clusters
+	// and to refresh the list of logical clusters periodically.
+	ListNames(ctx context.Context) ([]string, error)
+
+	// List returns a list of logical clusters.
```

Comment:
Does it return a list or a single LogicalClusterInfo?

Reply:
list of all names

Reply:
The comment refers to the Get() function. I think line 109 needs to be removed. @ezrasilvera .